webapp/views.py

```diff
@@ -173,6 +177,73 @@ def post_build():
     return flask.render_template("core/build.html", **context)
 
 
+def notify_build():
+    """
+    An endpoint to trigger an update about a build event to be sent.
+    This will usually be triggered by a webhook from Launchpad
+    """
+
+    # Verify contents
+    signature = hmac.new(
+        flask.current_app.config["SECRET_KEY"].encode("utf-8"),
+        flask.request.data,
+        hashlib.sha1,
+    ).hexdigest()
+
+    if "X-Hub-Signature" not in flask.request.headers:
+        return "No X-Hub-Signature provided\n", 403
+
+    if not hmac.compare_digest(
+        signature, flask.request.headers["X-Hub-Signature"].split("=")[1]
+    ):
+        print(signature)
+        return "X-Hub-Signature does not match\n", 400
```

Comment:
Remove print and indentation seems off.

Reply:
how dare you

Reply:
Removed the `print`. Don't know what you mean about indentation, seems fine.